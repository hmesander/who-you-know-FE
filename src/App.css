.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #283e4a;
  height: 30px;
  padding: 20px;
  display: flex;
  color: #fff;
  justify-content: flex-start;
}

.App-title {
  font-size: 1em;
}

.App-intro {
  font-size: large;
}

.login-button {
  color: #fff;
  height: 34px;
  line-height: 32px;
  width: 200px;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  font-weight: bold;
  background-color: #f6e312;
  border: 1px solid #e9ac1a;
  text-shadow: 0 1px 1px rgba(255,255,255,0.75);
  border-radius: 3px;
  color: #333;
  cursor: pointer;
  box-sizing: border-box;
  background-image: linear-gradient(to bottom, #f6e312 0%,#f9c80d 100%);
  text-shadow: 0 1px 1px rgba(255,255,255,0.75);
}

.login-button:hover {
  background-image: linear-gradient(to bottom, #eddb11 0%,#efcf14 35%,#f1bf18 65%,#f2b21b 100%);
  box-shadow: 0 1px 3px rgba(0,0,0,0.25);
}