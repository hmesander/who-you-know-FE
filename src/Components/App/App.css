.App {
  text-align: center;
}

body {
  background-color: #e7e7e7;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #283e4a;
  height: 30px;
  padding: 20px;
  display: flex;
  color: #fff;
  justify-content: space-between;
  align-items: center;
}

.App-title {
  font-size: 1.5em;
  background-color: none;
}

.App-intro {
  font-size: 1em;
  font-weight: 100;
  margin-right: 25px;
}

.login-container {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.buttons {
  color: #fff;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  font-weight: bold;
  background-color: #f6e312;
  border: 1px solid #e9ac1a;
  text-shadow: 0 1px 1px rgba(255,255,255,0.75);
  border-radius: 3px;
  color: #333;
  cursor: pointer;
  box-sizing: border-box;
  background-image: linear-gradient(to bottom, #f6e312 0%,#f9c80d 100%);
  text-shadow: 0 1px 1px rgba(255,255,255,0.75);
}

.buttons:hover {
  background-image: linear-gradient(to bottom, #eddb11 0%,#efcf14 35%,#f1bf18 65%,#f2b21b 100%);
  box-shadow: 0 1px 3px rgba(0,0,0,0.25);
}

.login-button {
  height: 24px;
  width: 100px;
}

.sort-cards-button, .play-deck-button {
  height: 34px;
  line-height: 32px;
  width: 200px;
} 