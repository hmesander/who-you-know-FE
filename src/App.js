import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import Controls from './Controls.js';
import Login from './Login.js';
import Sort from './Sort.js';
import './App.css';
import { authorizeUser } from './helpers/apiCalls.js';

class App extends Component {
  constructor() {
    super();

  }

  handleLogin = () => {
    console.log('click');
    this.props.history.push('/welcome');
  }

  handleSort = () => {
    console.log('sort');
    this.props.history.push('/sort');
  }

  handlePlayDeck = () => {
    console.log('play');
    this.props.history.push('/play');
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Who 
          you
          know.</h1>
        </header>
        <p className="App-intro">
          To get started, please login.
        </p>
        <button 
          className="login-button" 
          onClick={this.handleLogin}>
          Login
        </button>
        {/* <Route exact path='/' component={Login} /> */}
        <Route exact path='/welcome/' render={() => <Controls handleSort={this.handleSort} handlePlay={this.handlePlayDeck} />} />
        <Route path='/sort/' render={() => <Sort />} />
      </div>
    );
  }
}

export default withRouter(App);
