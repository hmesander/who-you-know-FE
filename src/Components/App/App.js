import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import Controls from '../Controls/Controls.js';
import Sort from '../Sort/Sort.js';
import PlayCards from '../PlayCards/PlayCards.js';
import './App.css';
import Auth from '../../Auth/Auth.js';
import history from '../../history';
import PropTypes from 'prop-types';
import { postTokens } from '../../helpers/apiCalls.js';
const auth = new Auth();

export class App extends Component {
  constructor() {
    super();

    this.state = {
      hash: '',
      idToken: '',
      accessToken: '',
      userId: ''
    };
  }

  async componentDidMount() {
    if (history.location.hash) {
      await this.setState({
        hash: history.location.hash
      });
      let { idToken, accessToken } = await this.parseURL(history.location.hash);
      this.setState({
        idToken,
        accessToken
      });
      let results = await postTokens(this.state.accessToken, this.state.idToken);
      this.setState({
        userId: results.id
      });
    } 
  }

  handleLogin = () => {
    auth.login();
  }

  handleLogout = () => {
    auth.logout();
    this.setState({
      hash: '', 
      userId: ''
    });
    this.props.history.push('/');
  }

  handleSort = () => {
    this.props.history.push('/sort');
  }

  handlePlayDeck = () => {
    this.props.history.push('/play');
  }

  parseURL = url => {
    let splitUrl = url.split('#access_token=');
    if (splitUrl.length > 1) {
      let accessToken = splitUrl[1].split('&expires_in=')[0];
      let idToken = splitUrl[1].split('&id_token=')[1];
      return { idToken, accessToken };
    }
  }

  render() {
    let display;
    if (this.state.hash !== '' || this.state.userId) {
      display = 
      <button
        className="buttons logout-button"
        onClick={this.handleLogout}>
          Logout
      </button>;
    } else {
      display = 
      <div className="login-container">
        <p className="App-intro">
          To get started, please login.
        </p>
        <button
          className="buttons login-button"
          onClick={this.handleLogin}>
          Login
        </button>
      </div>;
    }

    return (
      <div className="App">
        <header className="App-header">
          <div className='title-container'>
            <h1 className="App-title">Who 
              <span className='logo'>you</span> 
          know.</h1>
          </div>
          
          {display}
        </header>
        <Route path='/callback' render={() => <Controls handleSort={this.handleSort} handlePlay={this.handlePlayDeck} />} />
        <Route exact path='/sort' render={() => <Sort userId={this.state.userId} handlePlay={this.handlePlayDeck}/>} />
        <Route exact path='/play' render={() => <PlayCards userId={this.state.userId}/>} />
      </div>
    );
    
  }
}
export default withRouter(App);

App.propTypes = {
  history: PropTypes.string
};
