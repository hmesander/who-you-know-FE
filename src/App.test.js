import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';


describe('APP TESTS', () => {
  it('displays a login button if the user is not logged in', () => {
    const wrapper = shallow(<App history={''}/>, { disableLifecycleMethods: true });
    
    expect(wrapper.find('.login').length).toEqual(1);
  })

  it('displays a logout button if the user is logged in', () => {
    const wrapper = shallow(<App/>, { disableLifecycleMethods: true });
    wrapper.instance().setState({hash: 'taco'})
    
    expect(wrapper.find('.logout').length).toEqual(1);
  })
})