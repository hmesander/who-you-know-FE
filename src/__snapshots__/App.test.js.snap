// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP TESTS matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    history=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    Who you know.
          </h1>
</header>,
        <p
          className="App-intro"
>
          To get started, please login.
</p>,
        <button
          className="login-button login"
          onClick={[Function]}
>
          Login
</button>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="App-title"
>
            Who you know.
</h1>,
          "className": "App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Who you know.",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "Who you know.",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "To get started, please login.",
          "className": "App-intro",
        },
        "ref": null,
        "rendered": "To get started, please login.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
          "className": "login-button login",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Login",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        Who you know.
            </h1>
</header>,
          <p
            className="App-intro"
>
            To get started, please login.
</p>,
          <button
            className="login-button login"
            onClick={[Function]}
>
            Login
</button>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="App-title"
>
              Who you know.
</h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Who you know.",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "Who you know.",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "To get started, please login.",
            "className": "App-intro",
          },
          "ref": null,
          "rendered": "To get started, please login.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "className": "login-button login",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
